File: C++/sign-of-the-product-of-an-array.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    int arraySign(vector<int>& nums) {
        uint8_t flag = 0;
        for (const auto& x : nums) {
            if (x == 0) {
                return 0;
            }
            if (x < 0) {
                flag ^= 1;
            }
        }
        return flag % 2 ? -1 : 1;
    }
};

File: Python/sign-of-the-product-of-an-array.py
# Time:  O(n)
# Space: O(1)

class Solution(object):
    def arraySign(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        flag = 0
        for x in nums:
            if not x:
                return 0
            if x < 0:
                flag ^= 1
        return -1 if flag else 1
